src/app/ipv/middleware.ts

```diff
@@ -259,10 +234,11 @@ export const handleAppStoreRedirect: RequestHandler = (req, res, next) => {
       case PHONE_TYPES.ANDROID:
         return saveSessionAndRedirect(req, res, config.APP_STORE_URL_ANDROID);
       default:
-        throw new Error("Unrecognised phone type: " + specifiedPhoneType);
+        throw new BadRequestError(
+          "Unrecognised phone type: " + specifiedPhoneType,
+        );
```

Comment:
This changes the error from 500 -> 400 (although the user will see the same error page)